### DISCUSSION TITLE
Rule triggered by mqtt message / power rule that does not affect schedule

### DISCUSSION THREAD

**Discussion by ar2rh** [arendst/Tasmota#11022] [Q&A] [answered]
Hi 
Spent some time on it but cannot get it working.

I would like a rule to kick in when a message to power on the device comes through mqtt

I managed to get that working with the button 
`Rule1 ON button1#state DO Backlog Power1 %value%; RuleTimer1 60 ENDON   ON Rules#Timer=1 DO Power1 off ENDON`

in general I want the rule to kick in when the power is set to on manually and not by the schedule. If there is other way to accomplish that i would be open for suggestions

**Comment by sfromis** [comment]
The simpler way is PulseTime, but that would apply to all sources of turning power on. Your rules may have undesired effects if the scheduled command via MQTT comes in the 60-second interval. You could have the command via MQTT cancel the interval using `RuleTimer1 0`, assuming that the scheduled timing has to "win" over manual actions.

**Comment by barbudor** [comment]
If "rule to kick in when the power is set to on manually" means when the user press the button, you already have it.
What is the problem with your current rule?

**Reply by barbudor** [reply]
OK I understand now
Don't use MQTT power but use an event
You would fire the event through MQTT instead of firing Power
And have the same rule triggered by the event
Like 
Publish to `cmnd/yourdevice/event manualpower `
And rule `on event#manualpower do... `

**Comment by ar2rh** [comment]
thank you for you answers. maybe I should clarify what i want to accomplish

I have a timer set up (Configuration - configure timer) to switch on the light twice a day, lets say 9am-11am and 7pm to 9pm. that works great.
what i want to add is that when i press physical button (that works as per my first post) or toggle in the menu the light will stay up for 10 minutes (should 600 instead of 60 in my first post)
i can do that with
`Rule1 ON Power#state=1 DO Backlog Power1 %value%; RuleTimer1 600 ENDON   ON Rules#Timer=1 DO Power1 off ENDON`
and that works but the problem is that now when schedule kicks in the light turns of after 10minutes instead of 2 hours.
maybe there an "if " can be applied?
if powered on by schedule ignore the rule.

**Reply by sfromis** [reply]
If you want to have 3 channels of control (button, web UI and automation), this complicates things as the web UI button is not detectable by rules to control timing, or ignore the action, if you want to have light locked on when the automation has powered on. Button can be controlled using rules, and automations can fire off event commands. One way could be to have an extra virtual relay (on an unconnected gpio), open to web UI control, and then have a rule react to the power status of that "relay". You could also replicate the "primary" relay state to the fake one, to avoid it getting more complex than needed.

**Comment by ar2rh** [comment, marked as answer]
Thank you both for quick reply 

button is covered by this
`Rule1 ON button1#state DO Backlog Power1 %value%; RuleTimer1 60 ENDON ON Rules#Timer=1 DO Power1 off ENDON`

and i can get custom time set by the event here
`Rule2 ON event#manualpower DO Power1 on ENDON ON event#manualpower DO Var2 %value% ENDON ON event#manualpower DO RuleTimer1 %Var2% ENDON ON Rules#Timer=1 DO Power1 off ENDON`

that leaves the webui but I can live without it for now

**Reply by barbudor** [reply]
Why splitting the event into 3 individual rules instead of backlog?
You could also trigger te event from the button in order not to duplicate code

**Reply by ar2rh** [reply]
This is the only way i could get it to work.
Can you share the code and I will try it?

**Reply by barbudor** [reply]
Not tested as I don't have a complete test bed with me (just a nodemcu, no buttons, no wires)
As base, grouping the event on a single command should work
```
ON event#manualpower DO Backlog Power1 on; RuleTimer1 %value% ENDON
```

And you need only one trigger on the RuleTimer completion
No need to duplicate

For the Button, indeed as the state is only 2 (TOGGLE) I don't see a straight way for now to distinguish power off and power on

**Comment by ar2rh** [comment]
thanks that worked

`Rule2 ON event#manualpower DO Backlog Power1 on; RuleTimer1 %value% ENDON ON Rules#Timer=1 DO Power1 off ENDON`

the reason my code did not work was because i forgot to put "Backlog" in
